OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -26,7 +26,8 @@ public class LobbyLogic : ChromeLogic
 	{
 		static readonly Action DoNothing = () => { };
 
-		public MapPreview Map = MapCache.UnknownMap;
+		public MapPreview MapPreview = MapCache.UnknownMap;
+		public Map Map = null;
```

Comment:
You're missing null checks win a few places when accessing this field in `LobbyMapPreviewLogic.cs` and `LobbyLogic.cs` that cause crashes if the preview is slow to generate. This also isn't assigned in a thread-safe manner from the worker thread.